pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -530,15 +530,18 @@ def getData(self):
             if self.opts['logMode'][0]:
                 x = np.log10(x)
             if self.opts['logMode'][1]:
+                if 0 in y:
+                    y += 1
                 y = np.log10(y)
-            #if any(self.opts['logMode']):  ## re-check for NANs after log
-                #nanMask = np.isinf(x) | np.isinf(y) | np.isnan(x) | np.isnan(y)
-                #if any(nanMask):
-                    #self.dataMask = ~nanMask
-                    #x = x[self.dataMask]
-                    #y = y[self.dataMask]
-                #else:
-                    #self.dataMask = None
+                
+            if any(self.opts['logMode']):  ## re-check for NANs after log
```

Comment:
I think this will need a different solution as well. Maybe decouple these from this PR?